bindings/profilers/wall.cc

```diff
@@ -54,6 +54,29 @@ using namespace v8;
 
 namespace dd {
 
+int getTotalHitCount(const v8::CpuProfileNode* node) {
+  int count = node->GetHitCount();
+  auto child_count = node->GetChildrenCount();
+
+  for (int i = 0; i < child_count; ++i) {
+    count += getTotalHitCount(node->GetChild(i));
+  }
+  return count;
+}
+
+bool detectV8Bug(const v8::CpuProfile* profile) {
+  // Return true if either condition is false:
+  // - profile has at least one node with a non-zero hitcount
+  // - number of samples is strictly greater to total hitcount (meaning that
+  //   starting sample or some deopt events have been processed)
+  // If either condition is false, it implies that
+  // v8::SamplingEventsProcessor::ProcessOneSample loop is stuck for
+  // ticks_buffer_ or vm_ticks_buffer_.
```

Comment:
It's probably me, but I find it a bit convoluted to think in terms of the NAND operator ("true if either is false") :-) Can you rephrase it?
Here's a suggestion:
> When the profiler operates correctly, there'll be at least one node with positive hitcount and the number of samples will be strictly greater than the number of hits because they'll contain at least the starting sample and potentially some deoptimization samples. If these conditions don't hold, it implies that…